pkg/workloads/workload.go

```diff
@@ -0,0 +1,207 @@
+package workloads
+
+import (
+	"time"
+
+	"github.com/newrelic/newrelic-client-go/internal/serialization"
+	"github.com/newrelic/newrelic-client-go/pkg/errors"
+)
+
+// Workload represents a New Relic One workload.
+type Workload struct {
+	Account             *AccountReference        `json:"account,omitempty"`
+	CreatedAt           *serialization.EpochTime `json:"created_at,omitempty"`
+	CreatedBy           *UserReference           `json:"created_by,omitempty"`
+	Entities            []*EntityRef             `json:"entities,omitempty"`
+	EntitySearchQueries []*EntitySearchQuery     `json:"entitySearchQueries,omitempty"`
+	EntitySearchQuery   string                   `json:"entitySearchQuery,omitempty"`
+	GUID                *string                  `json:"guid,omitempty"`
+	ID                  *int                     `json:"id,omitempty"`
+	Name                *string                  `json:"name,omitempty"`
+	Permalink           *string                  `json:"permalink,omitempty"`
+	ScopeAccounts       *ScopeAccounts           `json:"scopeAccounts,omitempty"`
+	UpdatedAt           *serialization.EpochTime `json:"updated_at,omitempty"`
+}
+
+// AccountReference represents the account this workload belongs to.
+type AccountReference struct {
```

Comment:
I think this is common on most entities, and I've thought a few times about changing over to it from `accountId` so we get the name.  Might be worth moving this to `entities.AccountReference` so we can leverage it elsewhere.

Reply:
👍

Reply:
Entity has both:
```
{
  actor {
    entity(guid: "MjUwODI1OXxBUE18QVBQTElDQVRJT058MjA0MjYxNDEw") {
      account {
        id
        name
      }
      accountId
    }
  }
}
```
As well as the few interfaces I looked at.

Reply:
Moved to the `nerdgraph` package.